arcgisUM.py

```diff
@@ -0,0 +1,206 @@
+    instructorLog += '- - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n'
+    logger.debug("aCUTG: instructorLog 3: [{}]".format(instructorLog))
+
+    logger.info("addCanvasUsersToGroup: instructorLog: [{}]".format(instructorLog))
+    return instructorLog
+
+
+def getCurrentArcGISMembers(group, groupNameAndID):
+    groupAllMembers = {}
+
+    try:
+        groupAllMembers = group.get_members()
+    except RuntimeError as exception:
+        logger.error('Exception while getting users for ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            
+    groupUsers = groupAllMembers.get('users')
+    """:type groupUsers: list"""
+    return groupUsers
+
+def removeListOfUsersFromArcGISGroup(group, groupNameAndID, groupUsers):
+    """Remove only listed users from ArcGIS group."""
+
+    if len(groupUsers) == 0:
+        logger.info('No obsolete users to remove from ArcGIS Group {}'.format(groupNameAndID))
+        return None
+
+    logger.info('ArcGIS Users to be removed from ArcGIS Group [{}] [{}]'.format(groupNameAndID, ','.join(groupUsers)))
+    results = None
+    try:
+            results = group.removeUsersFromGroup(','.join(groupUsers))
+    except RuntimeError as exception:
+            logger.error('Exception while removing users from ArcGIS group "{}": {}'.format(groupNameAndID, exception))
+            
+    usersNotRemoved = results.get('notRemoved')
+    """:type usersNotRemoved: list"""
+    if usersNotRemoved:
+        logger.warning('Warning: Some or all users not removed from ArcGIS group {}: {}'.format(groupNameAndID, usersNotRemoved))
+        
+    return results
+
+
+def removeSomeExistingGroupMembers(groupTitle, group,instructorLog,groupUsers):
+    """Get list of ArgGIS users to remove from group and call method to remove them."""
+    results = ''
+    groupNameAndID = util.formatNameAndID(group)
+    logger.info('Found ArcGIS group: {}'.format(groupNameAndID))
+    instructorLog += 'Updating ArcGIS group: "{}"\n'.format(groupTitle)
+    
+    if not groupUsers:
+        logger.info('Existing ArcGIS group {} does not have users to remove.'.format(groupNameAndID))
+    else:
+        results = removeListOfUsersFromArcGISGroup(group, groupNameAndID, groupUsers)
+        
+    return instructorLog, results
+
+def createNewArcGISGroup(arcGIS, groupTags, groupTitle,instructorLog):
+    """Create a new ArgGIS group.  Return group and any creation messages."""
+   # group=None
+    
+    logger.info('Creating ArcGIS group: "{}"'.format(groupTitle))
+    instructorLog += 'Creating ArcGIS group: "{}"\n'.format(groupTitle)
+    try:
+        group = arcGIS.groups.create(groupTitle,groupTags)
+    except RuntimeError as exception:
+        logger.info('Exception while creating ArcGIS group "{}": {}'.format(groupTitle, exception))
```

Comment:
why not log it as `error` or `exception`

Reply:
Changed to log it with exception.